pkg/R/SQLContext.R

```diff
@@ -1,5 +1,130 @@
 # SQLcontext.R: SQLContext-driven functions
 
+infer_type <- function(x) {
+  if (is.null(x)) {
+    stop("can not infer type from NULL")
+  }
+  type <- switch(class(x),
+                 integer = "long",
```

Comment:
Just realized that the integer in R is only 32-bit (on 64bit machine), will change to use IntegerType in SparkR.